src/main/java/com/minecolonies/colony/jobs/AbstractJob.java

```diff
@@ -324,7 +324,7 @@ public void addTasks(@NotNull EntityAITasks tasks)
      *
      * @return your personal AI instance.
      */
-    public abstract AbstractAISkeleton generateAI();
+    public abstract AbstractAISkeleton<? extends AbstractJob> generateAI();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1452)